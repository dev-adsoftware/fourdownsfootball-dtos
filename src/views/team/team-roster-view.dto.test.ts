import { TeamRosterView } from '.';

describe('given: team roster data', () => {
  describe('when: I create a view', () => {
    it('then: TeamRosterView was returned', async () => {
      const view = new TeamRosterView().init({
        id: 'jest.team.id',
        sequence: '0',
        players: [],
      });
      expect(view).toEqual({ id: 'jest.team.id', sequence: '0', players: [] });
    });
  });
});
