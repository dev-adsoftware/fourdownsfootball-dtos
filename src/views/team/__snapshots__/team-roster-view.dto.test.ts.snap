// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given: missing data when: I initialize then: error matched snapshot 1`] = `[Error: TeamRosterView: id must be a string, sequence must be a number string, players must be an array, depthChart must be an object]`;

exports[`given: missing nested data when: I initialize then: error matched snapshot 1`] = `
[Error: TeamRosterView: id must be a string, sequence must be a number string, An instance of an object has failed the validation:
 - property players[0].id has failed the following constraints: isString 
 - property players[0].sequence has failed the following constraints: isNumberString 
, An instance of an object has failed the validation:
 - property depthChart.qb has failed the following constraints: isString 
 - property depthChart.rb has failed the following constraints: isString 
 - property depthChart.fb has failed the following constraints: isString 
 - property depthChart.wr has failed the following constraints: isString 
 - property depthChart.te has failed the following constraints: isString 
 - property depthChart.lt has failed the following constraints: isString 
 - property depthChart.lg has failed the following constraints: isString 
 - property depthChart.c has failed the following constraints: isString 
 - property depthChart.rg has failed the following constraints: isString 
 - property depthChart.rt has failed the following constraints: isString 
 - property depthChart.dt has failed the following constraints: isString 
 - property depthChart.de has failed the following constraints: isString 
 - property depthChart.ilb has failed the following constraints: isString 
 - property depthChart.olb has failed the following constraints: isString 
 - property depthChart.cb has failed the following constraints: isString 
 - property depthChart.s has failed the following constraints: isString 
 - property depthChart.k has failed the following constraints: isString 
 - property depthChart.p has failed the following constraints: isString 
 - property depthChart.kr has failed the following constraints: isString 
 - property depthChart.pr has failed the following constraints: isString 
 - property depthChart.ksp has failed the following constraints: isString 
 - property depthChart.krsp has failed the following constraints: isString 
 - property depthChart.prsp has failed the following constraints: isString 
 - property depthChart.psp has failed the following constraints: isString 
]
`;
